container/go/cmd/digester_index/digester_index.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This binary implements the ability to load a docker image index, calculate its image manifest sha256 hash and output a digest file.
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/bazelbuild/rules_docker/container/go/pkg/utils"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/bazelbuild/rules_docker/container/go/pkg/compat"
+	"github.com/bazelbuild/rules_docker/container/go/pkg/oci"
+)
+
+type applicationArgs struct {
+	Format         string
+	OutputFilePath string
+	Images         []imageArgs
+}
+
+type imageArgs struct {
+	Platform struct {
+		Os      string
+		Arch    string
+		Variant string
+	}
+	Config   string
+	Layers   []string
+	Tarball  string
+	Manifest string
+}
+
+func parseArgs(argv []string) (*applicationArgs, error) {
```

Comment:
This is not your fault, but an unfortunate consequence of working with the native flag package. Should we use something like [cobra](https://github.com/spf13/cobra)? It may make it easier to maintain.

Reply:
The use of cobra adds a new dependency, which is for me not really required 🤷‍♂️